src/neubot/dash_impl.hpp

```diff
@@ -0,0 +1,163 @@
+        },
+        "",
+        [=](Error error) {
+            if (error) {
+                std::cout << "Error: " << (int)error;
+                cb(error, nullptr);
+                return;
+            }
+
+            request_recv_response(
+                transport,
+                [=](Error error, Var<Response> res) {
+                    if (error) {
+                        std::cout << "Error: " << (int)error;
+                        cb(error, nullptr);
+                        return;
+                    }
+
+                    double new_times = mk::time_now();
+                    float length = res->body.length();
+                    double time_elapsed = new_times - saved_times;
+
+                    if (time_elapsed < 0) {
+                        std::cout << "Error: " << (int)error;
+                        cb(error, nullptr);
+                        return;
+                    }
+                    // TODO
+                    if (auth != "") {
+                        json::object_t result = {
+                            //{"connect_time", self.rtts[0]}
+                            //{"delta_user_time", delta_user_time}
+                            //{"delta_sys_time", delta_sys_time}
+                            {"elapsed", time_elapsed},        // OK
+                            {"elapsed_target", DASH_SECONDS}, // OK
+                            //{"internal_address", stream.myname[0]}
+                            {"iteration", iteration}, // OK
+                            //{"platform", sys.platform}
+                            {"rate", rate_kbit}, // OK
+                            //{"real_address", self.parent.real_address}
+                            //{"received", received}
+                            //{"remote_address", stream.peername[0]}
+                            //{"request_ticks", self.saved_ticks}
+                            //{"timestamp", mk::time_now}
+                            //{"uuid", self.conf.get("uuid")}
+                            //{"version", utils_version.NUMERIC_VERSION}
+                        };
+                        measurements->push_back(result);
+                    }
+
+                    double speed = double(length / time_elapsed);
+                    int s_k = (speed * 8) / 1000;
+
+                    std::cout << "DASH: [" << iteration << "/"
+                              << DASH_MAX_ITERATION << "] rate: " << rate_kbit
+                              << " Kbit/s, speed: " << s_k
+                              << " Kbit/s, elapsed: " << time_elapsed << " s\n";
+
+                    if (time_elapsed > DASH_SECONDS) {
+                        float relerr = 1 - (time_elapsed / DASH_SECONDS);
+                        s_k *= relerr;
+                        if (s_k < 0) {
+                            s_k = 100;
+                        }
+                    }
+
+                    loop_request(transport, s_k, cb, auth, measurements,
```

Comment:
I know I said differently in the past, but after #644 I prefer to avoid explicit recursion. could you wrap this invocation of `loop_request()` inside of `poller->call_soon([=]() { ... });`?